common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -80,15 +81,33 @@ protected[core] class Parameters protected[entity] (private val params: Map[Para
     params.keySet filter (params(_).init) map (_.name)
   }
 
+  protected[core] def getMap = {
+    params
+  }
   protected[core] def toJsArray = {
     JsArray(params map { p =>
-      if (p._2.init) {
-        JsObject("key" -> p._1.name.toJson, "value" -> p._2.value.toJson, "init" -> JsTrue)
-      } else JsObject("key" -> p._1.name.toJson, "value" -> p._2.value.toJson)
+      val init = p._2.init match {
+        case true => Some("init" -> p._2.init.toJson)
+        case _    => None
+      }
+      val encrypt = p._2.encryption match {
```

Comment:
I added some better tests and took your advice to tighten up this field. At first I though this might be where I stuck the IVs or other stuff the encryption scheme wanted to hold on to but I ended up just sticking that inside the raw value given that it's already a byte array.